File: app/src/main/java/com/legue/axel/lolsummonertool/wiki/fragment/WikiFragment.java
package com.legue.axel.lolsummonertool.wiki.fragment;


import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.google.android.material.tabs.TabLayout;
import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.legue.axel.lolsummonertool.R;
import com.legue.axel.lolsummonertool.SuperApplication;
import com.legue.axel.lolsummonertool.adapter.WikiFragmentAdapter;

import butterknife.BindView;
import butterknife.ButterKnife;


public class WikiFragment extends Fragment {

    private final static String TAG = WikiFragment.class.getName();
    private SuperApplication application;

    @BindView(R.id.vp_wiki)
    ViewPager vpWiki;
    @BindView(R.id.tl_wiki)
    TabLayout tlWiki;

    private WikiFragmentAdapter mWikiFragmentAdapter;

    public WikiFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_wiki_main, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        //pbLoading.setVisibility(View.GONE);

        //TODO : testing purpose => update code and move it at a better place
        application = (SuperApplication) getActivity().getApplication();

        mWikiFragmentAdapter = new WikiFragmentAdapter(getChildFragmentManager());
        vpWiki.setAdapter(mWikiFragmentAdapter);
        tlWiki.setupWithViewPager(vpWiki);

        vpWiki.setCurrentItem(0);
    }


}

File: app/src/main/java/com/legue/axel/lolsummonertool/view/wiki/fragment/WikiFragment.kt
package com.legue.axel.lolsummonertool.view.wiki.fragment


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.legue.axel.lolsummonertool.R
import com.legue.axel.lolsummonertool.SuperApplication
import com.legue.axel.lolsummonertool.view.adapter.WikiFragmentAdapter
import kotlinx.android.synthetic.main.fragment_wiki_main.*

class WikiFragment : Fragment() {

    companion object {

        private val TAG = WikiFragment::class.java.name
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_wiki_main, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        //pbLoading.setVisibility(View.GONE);

        //TODO : testing purpose => update code and move it at a better place
        val application = activity?.application as SuperApplication

        val mWikiFragmentAdapter = WikiFragmentAdapter(childFragmentManager)
        vp_wiki.adapter = mWikiFragmentAdapter
        tl_wiki.setupWithViewPager(vp_wiki)

        vp_wiki.currentItem = 0
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        retainInstance = true
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
    }

}
